src/services/request-presentation-service.ts

```diff
@@ -1,99 +1,135 @@
+import type { OutputOf } from 'codeco'
+import { LRUCache } from 'lru-cache'
 
-export type CommitPresentation = {
-  content: { path: string | undefined; prev: string; proof: string | undefined }
-  cid: string
-}
-
-export type NotCompletedRequestPresentation = {
-  createdAt: number
-  streamId: string
-  id: string
-  message: string
-  status: string
-  cid: string
-  updatedAt: number
-}
+const NAME_FROM_STATUS = {
+  [RequestStatus.REPLACED]: RequestStatusName.REPLACED,
+  [RequestStatus.FAILED]: RequestStatusName.FAILED,
+  [RequestStatus.PENDING]: RequestStatusName.PENDING,
+  [RequestStatus.PROCESSING]: RequestStatusName.PROCESSING,
+  [RequestStatus.READY]: RequestStatusName.READY,
+  [RequestStatus.COMPLETED]: RequestStatusName.READY,
+} as const
 
-export type CompletedRequestPresentation = NotCompletedRequestPresentation & {
-  anchorCommit: CommitPresentation
-}
-
-export type RequestPresentation = NotCompletedRequestPresentation | CompletedRequestPresentation
+const WITNESS_CAR_CACHE = 1000 // ~2MiB if one witness car is 2KiB
 
 /**
  * Render anchoring Request as JSON for a client to consume.
  */
 export class RequestPresentationService {
-  static inject = ['anchorRepository'] as const
+  static inject = ['anchorRepository', 'merkleCarService', 'witnessService'] as const
 
-  constructor(private readonly anchorRepository: IAnchorRepository) {}
+  private readonly cache = new LRUCache<string, CAR>({ max: WITNESS_CAR_CACHE })
 
-  /**
-   * Rich JSON of a request.
-   *
-   * @param request - Request to be rendered as JSON.
-   */
-  async body(request: Request): Promise<RequestPresentation> {
+  constructor(
+    private readonly anchorRepository: IAnchorRepository,
+    private readonly merkleCarService: IMerkleCarService,
+    private readonly witnessService: WitnessService
+  ) {}
+
+  async witnessCAR(anchor: AnchorWithRequest | null): Promise<CAR | null> {
+    if (!anchor) return null // TODO Add metric
+    const cacheKey = anchor.cid.toString()
+    const fromCache = this.cache.get(cacheKey)
+    if (fromCache) return fromCache
+    const merkleCAR = await this.merkleCarService.retrieveCarFile(anchor.proofCid)
+    if (!merkleCAR) return null // TODO Add metric
+    const witnessCAR = this.witnessService.buildWitnessCAR(anchor.cid, merkleCAR)
+    this.cache.set(cacheKey, witnessCAR)
+    return witnessCAR
+  }
+
+  async response(request: Request): Promise<CASResponse> {
     const status = request.status as RequestStatus
     switch (status) {
       case RequestStatus.COMPLETED: {
         const anchor = await this.anchorRepository.findByRequest(request)
+        const witnessCAR = await this.witnessCAR(anchor)
         // TODO: This is a workaround, fix in CDB-2192
-        const anchorCommit = {
-          cid: anchor ? anchor.cid.toString() : request.cid,
+        const anchorCommit: CommitPresentation = {
+          cid: anchor ? anchor.cid : CID.parse(request.cid),
           content: {
```

Comment:
not for this PR, but as a general question - should we remove this content object from the response entirely if it's not being used anywhere?

Reply:
Yep. That was done in js-ceramic codec. And, removed `content` from CAS codebase as well.